plugins/com.google.cloud.tools.eclipse.appengine.validation/src/com/google/cloud/tools/eclipse/appengine/validation/ElementProblem.java

```diff
@@ -59,10 +60,20 @@
    * A problem with no specific location in the document such as an element
    * that should be present but isn't.
    */
+  @VisibleForTesting
   ElementProblem(String message) {
     this(message, "org.eclipse.core.resources.problemmarker",
         IMarker.SEVERITY_WARNING, IMessage.NORMAL_SEVERITY, new DocumentLocation(0, 0), 0, null);
   }
+  
+  /**
+   * A problem with no specific location in the document such as an element
+   * that should be present but isn't.
+   */
+  ElementProblem(String message, String markerId) {
+    this(message, markerId,
```

Comment:
I wonder if we should be providing a (0,0) default rather than have the subclasses pass that down.

Reply:
0, 0 is deliberate here to get it on the document instead of a line. Not sure what you want to change.